Extension/src/LanguageServer/extension.ts

```diff
@@ -327,11 +327,35 @@ async function checkAndApplyUpdate(updateChannel: string): Promise<void> {
                 // then the .catch call will return a resolved promise
                 // Thusly, the .catch call must also throw, as a return would simply return an unused promise
                 // instead of returning early from this function scope
+                let config: vscode.WorkspaceConfiguration = vscode.workspace.getConfiguration();
+                let originalProxySupport: string = config.inspect<string>('http.proxySupport').globalValue;
+                while (true) { // Might need to try again with a different http.proxySupport setting.
+                    try {
+                        await util.downloadFileToDestination(buildInfo.downloadUrl, vsixPath)
+                            .catch(() => { throw new Error('Failed to download VSIX package'); });
```

Comment:
Why are you catching the promise and throwing a new error just to catch it in the try catch block?
You can remove the `.catch(() => { throw new Error('') });` and create the error object in the catch block on line 336.

Reply:
Okay, I was just following the pattern used by the existing code. I'll try changing it.